docs/docs/developers/docs/aztec-nr/framework-description/advanced/partial_notes.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Partial Notes
+sidebar_position: 1
+tags: [Developers, Contracts, Notes]
+description: How partial notes work and how they can be used.
+---
+
+import Image from "@theme/IdealImage";
+
+## What are Partial Notes?
+
+Partial notes are notes created with incomplete data, usually during private execution, which can be completed later with additional information that becomes available later, usually during public execution.
+
+Let’s say, for example, I have a `UintNote`:
+
+```rust
+pub struct UintNote {
+    owner: AztecAddress,    // Private field
+    randomness: Field,      // Private field
+    value: u128,            // Public field
+}
+```
+
+When creating the note locally, while in private execution, the `owner` is known, but the `value` potentially is not, e.g., it is some onchain dynamic variable. First, a **partial note** can be created during private execution that contains the `owner` and `randomness`, and then the note is *”completed”* to create a full note by later adding the `value` field, usually during public execution.
+
+<Image img={require("@site/static/img/partial-notes.png")} />
+
+## Use Cases
+
+Partial notes are useful when a e.g., part of the note struct is a value that depends on dynamic, public onchain data that isn't available during private execution, such as:
+
+- AMM swap prices
+- Current gas prices
+- Time-dependent interest accrual
+
+## Implementation
+
+All notes contain partial notes and use nested hash commitments. This is best explained using an example.
+
+### Note Structure Example
+
+Consider the `UintNote` structure again:
+
+```rust
+pub struct UintNote {
+    owner: AztecAddress,    // Private field
+    randomness: Field,       // Private field
+    value: u128,            // Public field
+}
+
+```
+
+### Two-Phase Commitment Process
+
+**Phase 1: Partial Commitment (Private Execution)**
+
+The private fields are committed during local execution:
+
+```rust
+let commitment = UintPartialNotePrivateContent { owner, randomness }
+    .compute_partial_commitment(storage_slot);
+
+fn compute_partial_commitment(self, storage_slot: Field) -> Field {
+    poseidon2_hash_with_separator(
+        self.pack().concat([storage_slot]),
+        GENERATOR_INDEX__NOTE_HASH,
+    )
+}
+
+```
+
+Here, we are creating a partial note commitment:
+
+```
+partial_commitment = H(partial_commitment, value)
```

Comment:
isnt this the complete commitment? the partial_commitment hashed with value is the complete commitment?
```suggestion
partial_commitment = H(storage_slot)
```